src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -553,6 +560,18 @@ trait Applications extends Compatibility { self: Typer =>
 
       fun1.tpe match {
         case ErrorType => tree.withType(ErrorType)
+        case TryDynamicCallType =>
+          tree match {
+            case Apply(Select(qual, name), args) if !isDynamicMethod(name) =>
+              if (untpd.isWildcardStarArgList(args)) {
+                ctx.error(name.toString + " does not support passing a vararg parameter", tree.pos)
+                tree.withType(ErrorType)
```

Comment:
The current implementation does what the scalac implementation does. Im not sure why the restriction was there in the first place. It could be just because it is harder to handle with `applyDynamicNamed`.